R/calcPastureSuit.R

```diff
@@ -29,88 +29,80 @@ calcPastureSuit <- function(climatetype = "MRI-ESM2-0:ssp126",
+
+  cellPrep <- calcOutput("LPJmLClimateInput",
+    climatetype = climatetype,
+    variable = "precipitation:monthlySum",
+    stage = stage,
+    lpjmlVersion = lpjml,
+    aggregate = FALSE
+  )
+
+
+  cellPet <- calcOutput(
+    type = "LPJmL_new", climatetype = climatetype,
+    subtype = "mpet",
+    stage = stage,
+    version = lpjml,
+    aggregate = FALSE
+  )
+
+
+  yearsCellPet <- intersect(getYears(cellPet), findset("time"))
+  yearsCellPrep <- intersect(findset("time"), getYears(cellPrep))
+  years <- intersect(yearsCellPet, yearsCellPrep)
+  cellPrep <- dimSums(cellPrep[, years, ], dim = 3)
+  cellPet <- dimSums(cellPet[, years, ], dim = 3)
 
+  # Cell area calculation
+  land <- calcOutput("LanduseInitialisation", input_magpie = TRUE,
+                     aggregate = FALSE, cellular = TRUE, cells = cells,
+                     years = "y1995", round = 6)
+  landArea <- setYears(dimSums(land, dim = 3), NULL)
 
-  cellPet    <- calcOutput(type = "LPJmL_new", climatetype = climatetype,
-                           subtype   = "mpet",
-                           stage     = stage,
-                           version   = lpjml,
-                           aggregate = FALSE)
-
-  yearsCellPet      <- intersect(getYears(cellPet), findset("time"))
-  yearsCellPrep     <- intersect(findset("time"), getYears(cellPrep))
-  years             <- intersect(yearsCellPet, yearsCellPrep)
-  cellPrep          <- dimSums(cellPrep[, years, ], dim = 3)
-  cellPet           <- dimSums(cellPet[, years, ], dim = 3)
+  cellorder <- population
+  cellorder <- match(dimnames(cellorder)$x.y.iso, dimnames(landArea)$x.y.iso)
 
-  # Cell area calculation
-  landArea <- calcOutput("LandArea", aggregate = FALSE) / 100 # transformed to km^2
+  # Reorder the dimensions of landArea to match Population
+  landArea <- landArea[cellorder, , ]
+  getCells(landArea) <- getCells(population)
 
   # population density
-  popDensity <- (population * 1e6) / landArea # population density in number of people per km2
+  popDensity <- (population * 1e6) / landArea * 10000 # population density in number of people per km2
   popDensity[is.infinite(popDensity)] <- 0
   popDensity[is.nan(popDensity)] <- 0
 
   years <- intersect(getYears(popDensity), getYears(cellPrep))
 
   aridity <- cellPrep[, years, ] / cellPet[, years, ]
   aridity[is.infinite(aridity) | is.nan(aridity)] <- 0
+
   # 0.5 aridity threshold for managed pastures. Same from HYDE 3.2.
   aridity[aridity < 0.5] <- 0
   aridity[aridity >= 0.5] <- 1
 
   # pasture suitability check
   pastureSuit <- aridity
   popDensity <- popDensity[, getYears(pastureSuit), ]
+
+  cellorder <- match(dimnames(pastureSuit)$x.y.iso, dimnames(popDensity)$x.y.iso)
+  popDensity <- popDensity[cellorder]
   pastureSuit[popDensity < 5] <- 0 # 5 hab km2 population threshold for managed pastures. Same from HYDE 3.2.
 
   pastureSuitArea <- (pastureSuit * landArea * 100) / 1e6 # (from km2 (x100) to mha (/1e6))
```

Comment:
Please check again if this unit transformation is correct. I think maybe not.
PastureSuit should be either 0 or 1, Landarea already in mio. ha see LN:65 (you load it as mio. ha) and in LN:75 you do not transform the variable LandArea it self to km^2.

Reply:
Hi Kistina, this line of code does not work as expected because the order of the cells between  popDensity and pastureSuit where diffrent. 
I have updated this replacing by a multiplication that gurantess the the cells are multiplied in the correct order.